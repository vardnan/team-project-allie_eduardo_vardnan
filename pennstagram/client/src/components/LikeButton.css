#liked {
    color: white;
    background-color: #ff5571;
    border-radius: 3px;
}

.likeButton {
    margin-left: 5px;
    margin-bottom: 5px;
}

.buttonBox {
    display: flex;
    height: 45px;
    align-items: center;
}

.likeNumber {
    margin-left: 8px;
}