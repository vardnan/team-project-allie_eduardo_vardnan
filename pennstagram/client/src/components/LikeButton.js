import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './LikeButton.css';
import { toggleLike } from '../api/toggleLike';

function LikeButton({ currentUser, likes, postId }) {
  // add line testing if postLikes are recieved from activity feed
  const [isLiked, setIsLiked] = useState(likes.includes(currentUser.username));

  const likeHandler = async () => {
    try {
      if (isLiked) {
        setIsLiked(false);
        likes.pop();
        await toggleLike(postId, currentUser, false); // API call to unlike the post
      } else {
        setIsLiked(true);
        likes.push(currentUser.username);
        await toggleLike(postId, currentUser, true); // API call to like the post
      }
    } catch (error) {
      // eslint-disable-next-line
      console.error('Error toggling like:', error);
      // Handle the error if needed
    }
  };

  return (
    <div className='buttonBox'>
      <button
        type="button" // Add the type attribute with the value "button"
        id={isLiked ? 'liked' : 'unliked'}
        className="likeButton"
        onClick={likeHandler}
        data-cy="like-button"
      >
        {isLiked ? 'Unlike' : 'Like'}
      </button>
      <p className='likeNumber'>{likes.length}</p>
    </div>
  );
}

LikeButton.propTypes = {
  currentUser: PropTypes.object.isRequired,
  likes: PropTypes.array.isRequired,
  postId: PropTypes.string.isRequired,
};

export default LikeButton;
