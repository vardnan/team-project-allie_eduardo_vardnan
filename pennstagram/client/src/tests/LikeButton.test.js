import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import LikeButton from '../components/LikeButton';

// rendering testing
test('Testing that the Unlike state is set correctly', () => {
  const props = {
    currentUser: 'Alice',
    likes: ['Alice'],
    setLikes: jest.fn(),
  };

  const { getByText } = render(<LikeButton {...props} />);
  expect(getByText('Unlike')).toBeInTheDocument();
});

// snapshot testing
test('Like button matches snapshot', () => {
  const props = {
    currentUser: 'Alice',
    likes: ['Bob'],
    setLikes: jest.fn(),
  };

  const component = renderer.create(<LikeButton {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// event testing
test('The like button is toggling correctly', () => {
  const mockSetLikes = jest.fn();
  const props = {
    currentUser: 'Alice',
    likes: ['Bob'],
    setLikes: mockSetLikes,
  };

  const { getByText } = render(<LikeButton {...props} />);
  const button = getByText('Like');
  fireEvent.click(button);
  expect(mockSetLikes).toHaveBeenCalledWith(['Alice']);
});
