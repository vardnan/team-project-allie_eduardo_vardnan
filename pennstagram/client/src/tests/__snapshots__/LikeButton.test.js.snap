// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Like button matches snapshot 1`] = `
<div>
  <button
    className="likeButton"
    id="unliked"
    onClick={[Function]}
  >
    Like
  </button>
</div>
`;
